sos/cleaner/__init__.py

```diff
@@ -769,15 +769,21 @@ def obfuscate_file(self, filename, short_name=None, arc_name=None):
         if not os.path.islink(filename):
             # don't run the obfuscation on the link, but on the actual file
             # at some other point.
-            self.log_debug(f"Obfuscating {short_name or filename}",
-                           caller=arc_name)
-            tfile = tempfile.NamedTemporaryFile(mode='w', dir=self.tmpdir)
             _parsers = [
                 _p for _p in self.parsers if not
                 any(
                     _skip.match(short_name) for _skip in _p.skip_patterns
                 )
             ]
+            if not _parsers:
+                self.log_debug(
+                    f"Skipping obfuscation of {short_name or filename} due to "
+                    f"matching file skip pattern"
+                )
```

Comment:
Great! I missed log like this (and got confusing log `Obfuscating <filename>` instead) when discovering the reproducer for the issue.